### ISSUE TITLE
pyls-lsp and pylsp-lsp calls `flake` despite the plugin being disabled

### ISSUE THREAD

**Issue by Hi-Angel** [emacs-lsp/lsp-mode#3454]
- [X] I am using the latest version of `lsp-mode` related packages.
- [X] I checked [FAQ](https://emacs-lsp.github.io/lsp-mode/page/faq/) and [Troubleshooting](https://emacs-lsp.github.io/lsp-mode/page/troubleshooting/) sections
- [ ] You may also try reproduce the issue using clean environment using the following command: `M-x lsp-start-plain`

### Bug description

Starting `lsp` in a python file gives me tons of annoying warnings coming from `flake`, even though `lsp-pyls-plugins-pyflakes-enabled` is set to `nil`. I have no idea how to stop `pyls-lsp` from doing that.

I use `mypy` instead, and these two does not play well together. I even removed `python-flake` completely from my system, but by some magic it still produces flycheck warnings into my buffer. I can identify it's `flake` due to prefix `E261` on one of the warnings.

### Steps to reproduce

1. Make sure `python-language-server` is installed
2. ```
    λ cat test.py
   print('Hello') # this comment will cause a flake E261 warning
    λ cat ~/.emacs
   (package-initialize)
   (require 'lsp)
   (setq lsp-pyls-plugins-pyflakes-enabled nil)
    λ emacs test.py --eval '(lsp)'
    ```
3. A emacs window will appear with `test.py` file opened and `lsp` running. Look at the python comment.


### Expected behavior

The python comment will not be highlighted by `flake`, since it is disabled.

### Which Language Server did you use?

`pyls`.

### OS

Linux

### Error callstack

_No response_

### Anything else?

_No response_

**Comment by Hi-Angel** [comment]
As an experiment, I tried removing everything flake-related from the `lsp-pyls.el`, and that still didn't work.

**Comment by Hi-Angel** [comment]
Hm, this is weird. I removed all mentions of `flake` from `lsp-mode` elpa directory *(and all `*.elc` files)*, and still upon restarting Emacs with a python file I get those warnings o.O

**Comment by yyoncho** [comment]
can you switch to `pylsp`? `pyls` is deprecated.

**Comment by Hi-Angel** [comment]
> can you switch to `pylsp`? `pyls` is deprecated.

I tried 4-5 months ago, but stumbled upon some `pylsp`-specific problems *(can't remember details at this point)* and decided to go back for the time being, since pyls works for me.

**Comment by yyoncho** [comment]
AFACS the name of the property is `lsp-pyls-plugins-flake8-enabled`

**Comment by Hi-Angel** [comment]
So, I migrated to `pylsp`, and it has same problem. I set:

```
        lsp-pylsp-plugins-flake8-enabled nil
        lsp-pylsp-plugins-pyflakes-enabled nil
```

to no avail. At this point I am not sure if it's a problem with lsp-mode not passing through the parameters, or with the `pylsp` itself. I suspect `lsp-mode` here, because over at `python-lsp-server` nobody's complaining, it seems, about flakes.

**Comment by Hi-Angel** [comment]
Is there any way to print what configuration values end up being passed to pylsp?

**Comment by yyoncho** [comment]
lsp-pylsp-plugins-flake8-enabled is different from lsp-pyls-plugins-flake8-enabled. Set lsp-print-io to t and then use M-x lsp-workspace-show-log

**Comment by Hi-Angel** [comment]
> lsp-pylsp-plugins-flake8-enabled is different from lsp-pyls-plugins-flake8-enabled

Right, since I migrated to `pylsp`, I set the pylsp one *(actually, I set both pylsp and pyls versions of the two variables to `nil`)*.

> Set lsp-print-io to t and then use M-x lsp-workspace-show-log

This prints server reply. What I'm interested in is to see what configuration `lsp-mode` passed to the server as part of `lsp-client-settings`, because currently I suspect it might not be passing `false` one of `pyls.plugins.pyflakes.enabled` or `pyls.plugins.flake8.enabled`.

**Comment by FabriceSalvaire** [comment]
Look at https://emacs-lsp.github.io/lsp-mode/page/troubleshooting and use `(setq lsp-log-io t)` .
Then look at the log.
It solved my nightmare.

**Comment by Hi-Angel** [comment]
Thank you for the suggestion, that did help! Even though I didn't manage to make it work, but at least now I suspect it is actually a bug in `pylsp` per se. While setting the variable before enabling the mode and then executing `lsp`, I now can see in the `M-x lsp-workspace-show-log` buffer the JSON that gets sent to pylsp for initialization. It has flakes plugin off *(per my settings)*, and I also found out a couple more possibly influencing plugins that I disabled as well. I can see now that the JSON has everything disabled, however `pylsp` stubbornly keeps sending a flakes-related warning. So it's likely a bug in `pylsp`, I don't think there's anything `lsp-mode` can do in this case.

**Comment by Hi-Angel** [comment]
Huh, so, setting this variable now makes it work:

```lisp
(setq lsp-pylsp-plugins-pydocstyle-enabled nil)
```

However I should mention that it did not work before, when I was writing the comment, and I restarted Emacs multiple times back then, testing other stuff as well. So whether it works is somehow sporadical. However, `lsp-mode` sends property `disabled` for that plugin correctly to the server *(I mean, I checked that back when I wrote the previous comment)*, so if that doesn't work that'd be still the `pylsp` bug. So, anyway, with this variable set I hope problem won't appear again.
